### ISSUE TITLE
popup items get stuck

### ISSUE THREAD

**Issue by aspauldingcode** [FelixKratz/SketchyBar#638]
I have subscribed items to on-hover as mouse.entered, mouse.entered.global, mouse.exited, mouse.exited.global and I sometimes have issues where the popups get stuck and don't update.

sketchybar-v2.21.0
macOS-14.7

**Comment by MisutaaAsriel** [comment]
Same here, macOS 15.1 2.21.0

**Comment by MisutaaAsriel** [comment]
On further testing, it seems **all** events are randomly getting stuck. Sometimes switching apps doesn't register and I end up with a menu bar that has no menu items. mouse events seem to just be the most interacted with and therefor most noticeable.

**Comment by MisutaaAsriel** [comment]
Yeah, the more I use this the more I am noticing that events are not always firing correctly. All events, including app switching. I have a custom bar with the menu items, and it'll randomly just… be empty when switching apps. It's like it fired too early. Sometimes `mouse.exited` and `mouse.exited.global` won't fire. Etc.

**Comment by aspauldingcode** [comment]
Closing. Clone of https://github.com/FelixKratz/SketchyBar/issues/613.
